### ISSUE TITLE
ModuleNotFoundError: No module named 'pyfftw.pyfftw'

### ISSUE THREAD

**Issue by willamgld** [pyFFTW/pyFFTW#291]
Hello, 

I have struggled and googled for hours, and I am unable to import pyfftw on my Windows 10 machine running Anaconda with ipython 3.7. 
```
% import pyfftw

ModuleNotFoundError                Traceback (most recent call last)
<ipython-input-13-1dffee0b3d50> in <module>
----> 1 import pyfftw

C:\ProgramData\Anaconda3\lib\site-packages\pyfftw\__init__.py in <module>
     16 import os
     17 
---> 18 from .pyfftw import (
     19         FFTW,
     20         export_wisdom,

ModuleNotFoundError: No module named 'pyfftw.pyfftw'
```
I have tried all the `conda install` commands recommended  on [Anaconda.com](https://anaconda.org/conda-forge/pyfftw) and local installs from tarballs. I tried versions 0.10, 0.11 and 0.12 of pyfftw. I have the prerequisite packages installed (numpy and cython), and I have fftw's  win-x64 DLLs in my `PATH`. What am I missing?

**Comment by grlee77** [comment]
Can you run `conda list fftw` and paste the output here? It should look similar to the below (although the exact Build will differ as the example below was on linux):
```
# Name                    Version                   Build  Channel
fftw                      3.3.8           nompi_h7f3a6c3_1109    conda-forge
pyfftw                    0.12.0           py37hc3ef4f8_1    conda-forge
```

I just tried on Python 3.7, pyFFTW 0.12 on an old Windows 7 machine and it imported okay.

**Comment by willamgld** [comment]
```
(base)> conda list fftw
# packages in environment at C:\ProgramData\Anaconda3:
#
# Name                    Version                   Build  Channel
fftw                      3.3.8                he774522_3
pyfftw                    0.12.0           py37h2c60a8d_1    <unknown>
```
It says channel unknown is because I most recently installed it from a tarball downloaded from [conda-forge](https://anaconda.org/conda-forge/pyfftw/files) .

**Comment by HaHeho** [comment]
I have noticed something dysfunctional behavior with the version from `conda-forge` on Win64 as well and ended up installing via `pip`.

I've tried again, ending up with
```
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
certifi                   2020.4.5.1       py38h32f6830_0    conda-forge
fftw                      3.3.8           nompi_hbc43ac9_1110    conda-forge
intel-openmp              2020.1                      216
libblas                   3.8.0                    15_mkl    conda-forge
libcblas                  3.8.0                    15_mkl    conda-forge
liblapack                 3.8.0                    15_mkl    conda-forge
mkl                       2020.0                      166
numpy                     1.18.4           py38h72c728b_0    conda-forge
openssl                   1.1.1g               he774522_0    conda-forge
pip                       20.1               pyh9f0ad1d_0    conda-forge
pyfftw                    0.12.0           py38hd17e47c_1    conda-forge
python                    3.8.2           h5fd99cc_7_cpython    conda-forge
python_abi                3.8                      1_cp38    conda-forge
setuptools                46.2.0           py38h32f6830_0    conda-forge
sqlite                    3.30.1               hfa6e2cd_0    conda-forge
vc                        14.1                 h869be7e_1    conda-forge
vs2015_runtime            14.16.27012          h30e32a0_2    conda-forge
wheel                     0.34.2                     py_1    conda-forge
wincertstore              0.2                   py38_1003    conda-forge
```

At least `import pyfftw` seems to work without crashing now. I have not verified any functionality though.

**Comment by HaHeho** [comment]
Import also works after installing e.g. Python and Numpy from `conda main` and pyfftw via `conda-forge`:
```
blas                      1.0                         mkl
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
certifi                   2020.4.5.1       py38h32f6830_0    conda-forge
fftw                      3.3.8           nompi_hbc43ac9_1110    conda-forge
icc_rt                    2019.0.0             h0cc432a_1
intel-openmp              2020.1                      216
mkl                       2020.1                      216
mkl-service               2.3.0            py38hb782905_0
mkl_fft                   1.0.15           py38h14836fe_0
mkl_random                1.1.0            py38hf9181ef_0
numpy                     1.18.1           py38h93ca92e_0
numpy-base                1.18.1           py38hc3f5095_1
openssl                   1.1.1g               he774522_0    conda-forge
pip                       20.0.2                   py38_3
pyfftw                    0.12.0           py38hd17e47c_1    conda-forge
python                    3.8.2               he1778fa_13
python_abi                3.8                      1_cp38    conda-forge
setuptools                46.2.0                   py38_0
six                       1.14.0                   py38_0
sqlite                    3.31.1               h2a8f88b_1
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.16.27012          hf0eaf9b_1
wheel                     0.34.2                   py38_0
wincertstore              0.2                      py38_0
zlib                      1.2.11               h62dcd97_4
```

**Comment by willamgld** [comment]
> I have noticed something dysfunctional behavior with the version from `conda-forge` on Win10 as well and ended up installing via `pip`.

Can you please detail the steps you took to install pyfftw using pip? Did you run `pip install pyfftw` within an Anaconda Command Prompt, or did you use a non-Anaconda Python distribution altogether? I would very much prefer to stay with Anaconda. I tried pip in Anaconda, but it couldn't build a wheel because it couldn't find MS VS C++ 14.0, even though I installed and reinstalled build tools for [2015](http://go.microsoft.com/fwlink/?LinkId=691126&fixForIE=.exe), [2017 and 2019](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&rel=16). 

Right now, with pyfftw v. 0.12 already installed in my base environment from a local tarball, when I issue 
`(base) >conda install -c conda-forge pyfftw` , I get (abbreviated)
```
...
The environment is inconsistent, please check the package plan carefully
...
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - pyfftw -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0']

Your python: python=3.7
```
Which, basically, says that pyfftw is incompatible with Python 3.7. 
OK, so I created a new environment with python 3.8.2, numpy and cython , and the same install command in that environment now fails to find the package:
```
PackagesNotFoundError: The following packages are not available from current channels:

  - pyfftw
```
My best guess is that this is still some version mismatch that makes the package invisible. 
I am completely stumped. I need to be working on someone else's code, and I cannot even get past the package install... Any help is highly appreciated, thanks.

**Comment by HaHeho** [comment]
> Import also works after installing e.g. Python and Numpy from `conda main` and pyfftw via `conda-forge`:

As I said, the two versions I've tested were both based on conda. Starting from a new environment seems like a good choice in any case. This always lead to an environment where I was able to import `pyfftw`:
`conda create -n pyfftw_test`
`conda activate pyfftw_test`
`conda install python numpy`
`conda install -c conda-forge pyfftw`

You could skip the `install python numpy` command, but would then end up with versions of these packages from `conda-forge` (due to the `pyfftw` install command). This can have a major influence on performance (not sure how relevant that is in non-real-time applications), due to different versions and optimizations of `numpy` depending on your system architecture. Although, this should not really be related to your problem, I suppose (most straight forward solution: on Intel systems go with the installation of `numpy` via `conda-main`, not `conda-forge`).

**Comment by willamgld** [comment]
I ran the above commands verbatim, everything worked up until the last command that failed with the same message:
```
(pyfftw_test) > conda install -c conda-forge pyfftw
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pyfftw

Current channels:

  - https://conda.anaconda.org/conda-forge/win-32
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/win-32
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-32
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-32
  - https://repo.anaconda.com/pkgs/msys2/noarch
...
```
Here is the list of packages in the test environment:
```
(pyfftw_test) C:\Users\wgoldshlag>conda list
# packages in environment at C:\ProgramData\Anaconda3\envs\pyfftw_test:
#
# Name                    Version                   Build  Channel
blas                      1.0                         mkl
ca-certificates           2020.1.1                      0
certifi                   2020.4.5.1               py38_0
icc_rt                    2019.0.0             h0cc432a_1
intel-openmp              2020.1                      216
mkl                       2020.1                      216
mkl-service               2.3.0            py38he774522_0
mkl_fft                   1.0.15           py38h14836fe_0
mkl_random                1.1.0            py38hf9181ef_0
numpy                     1.18.1           py38h93ca92e_0
numpy-base                1.18.1           py38hc3f5095_1
openssl                   1.1.1g               he774522_0
pip                       20.0.2                   py38_3
python                    3.8.2               he1778fa_13
setuptools                46.2.0                   py38_0
six                       1.14.0                   py38_0
sqlite                    3.31.1               h2a8f88b_1
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.16.27012          hf0eaf9b_1
wheel                     0.34.2                   py38_0
wincertstore              0.2                      py38_0
zlib                      1.2.11               h33f27b4_4
```
Python are numpy are not from conda-forge, nor have they ever been for me in any of my other environments.

**Comment by HaHeho** [comment]
One notable difference is that you have `win-32` in your conda channels, instead of `win-64` in my case. Are you on 32bit Windows? No idea if that actually makes a difference here or how this could be tested on a respective 32 or 64 bit system.

**Comment by willamgld** [comment]
I am on a 64-bit Intel / Win 10 system, but I have a 32-bit Anaconda installed. I can't remember exactly why I made that choice, but I bet it was because of some software incompatibility. I'll see if I can find another machine to test it without disturbing my current setup. Could anyone weigh in on whether this could be the real cause?

**Comment by HaHeho** [comment]
Installing 32 bit and 64 bit versions of Miniconda seems to be possible and different environments should be manageable. Maybe check out this https://stackoverflow.com/questions/33709391/using-multiple-python-engines-32bit-64bit-and-2-7-3-5

**Comment by willamgld** [comment]
I confirmed that the issue was the bitness of the Python installation. I switched to 64 bit, and pyfftw installed right away. However, I remembered why I had the 32-bit Anaconda the hard way: it broke my gVim - Python IDE because of bitness mismatch, and I ended up spending a lot of time hunting down the 64-bit gVim binaries and reinstalling some of its plugins.
PS. I didn't see your comment about Miniconda until it was all finished: I will check it out, thanks!
